### PR TITLE
Set C++ to 17, and a few CMake fixes for Linux

### PR DESCRIPTION
**Pull request by th1000s**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -14,14 +14,19 @@ PROJECT(Descent3)
...
+    SET(EXTRA_CXX_FLAGS "-Wno-address-of-temporary")
+  ELSE()
+    SET(BITS "-m32")
+    SET(EXTRA_CXX_FLAGS "-fpermissive")
+  ENDIF()
+
   SET(CMAKE_CXX_COMPILER "g++")
-  SET(CMAKE_CXX_FLAGS "-O0 -g -Wno-write-strings -Wno-multichar -Wno-address-of-temporary")
-  SET(CMAKE_C_FLAGS "-O0 -g")
+  SET(CMAKE_CXX_FLAGS "-O0 -g -std=c++14 -Wno-write-strings -Wno-multichar ${BITS} ${EXTRA_CXX_FLAGS}")
```
Comment (winterheart) [inline CMakeLists.txt:26]: You need this done like this
```
set(CMAKE_CXX_STANDARD 14)
set(CMAKE_CXX_STANDARD_REQUIRED ON)
set(CMAKE_CXX_EXTENSIONS OFF)
```
But there some issues in with CMake compatibility level. Also, as agreed in #4, there should be C++17 standard support.

File: CMakeLists.txt [hunk 2]
```diff
@@ -14,14 +14,19 @@ PROJECT(Descent3)
 IF (UNIX)
 SET (D3_GAMEDIR "~/Descent3/")
 
+  IF (APPLE)
```
Comment (winterheart) [inline CMakeLists.txt:17]: I doubt that in Apple there may 64bit code generated. Also, -m32 should be added via compiler toolchain, not directly in CMake project file.

**Review by winterheart — COMMENTED** [review]
Overall, there need to CMake project get updated, as there some old tricks made in 2.x era. I'd suggest to rewrite CMakeLists.txt to 3.11 compatibility level to get rid some of these quirks.

**Comment by Arcnor** [comment]
I'd actually wait to see the outcome of #9 before merging this and similar PRs, given that Ryan Gordon had a much more "modern" port for Linux and macOS, just my 2c.

File: CMakeLists.txt (hunk 1 above)
Comment (th1000s) [reply CMakeLists.txt]: Done, plus a few lines of code adapted. And indeed, that is just functional cmake to get the code to compile.

File: CMakeLists.txt (hunk 2 above)
Comment (th1000s) [reply CMakeLists.txt:22]: Yes, `-m32` did not quite work, removed entirely.

**Comment by Lgt2x** [comment]
Merging, thanks for your contribution !

**Pull request merged**